packages/flutter/test/foundation/memory_allocations_test.dart

```diff
@@ -160,20 +162,97 @@ void main() {
     _checkSdkHandlersNotSet();
     expect(ma.hasListeners, isFalse);
   });
+
+  testWidgets('State dispatch events in debug mode', (WidgetTester tester) async {
+    bool stateCreated = false;
+    bool stateDisposed = false;
+
+    void listener(ObjectEvent event) {
+      if (event is ObjectCreated && event.object is State) {
+        stateCreated = true;
+      }
+      if (event is ObjectDisposed && event.object is State) {
+        stateDisposed = true;
+      }
+    }
+    ma.addListener(listener);
+
+    await tester.pumpWidget(const _TestStatefulWidget());
+    await tester.pumpWidget(const SizedBox.shrink());
```

Comment:
maybe assert in between the pumps that the state was created but not disposed yet?